examples/ota_from_flash.rs

```diff
@@ -0,0 +1,35 @@
+extern crate esp_ota;
+use log::*;
+
+const FIRMWARE_ID: &str = "alpha";
+
+const NEW_APP: &[u8] = include_bytes!("../beta.bin");
+
+fn main() {
+    esp_idf_svc::sys::link_patches();
+    esp_idf_svc::log::EspLogger::initialize_default();
+
+    info!("FIRMWARE ID : {}", FIRMWARE_ID);
+    if FIRMWARE_ID == "alpha" {
```

Comment:
What is this `if FIRMWARE_ID = ...` condition supposed to teach the reader? It will by definition always be `"alpha"` since it's a constant defined in the same file/firmware.

I understand that this example becomes a bit contrived in general given that it bundles the new firmware right into itself... Maybe the http example would be more educational, however also a bit more complex.

Reply:
The condition' not that important, it's more an educational exercise to show how rollback detection might be implemented.